src/client/java/teammates/client/scripts/DataMigrationForFeedbackResponseCommentSearchDocumentDateFormat.java

```diff
@@ -0,0 +1,166 @@
+    private static final JsonParser jsonParser = new JsonParser();
+
+    public static void main(String[] args) throws IOException {
+        new DataMigrationForFeedbackResponseCommentSearchDocumentDateFormat().doOperationRemotely();
+    }
+
+    @Override
+    protected void doOperation() {
+        List<FeedbackResponseCommentAttributes> commentsToFix = analyzeAndGetAffectedComments();
+        fixDocuments(commentsToFix);
+    }
+
+    private List<FeedbackResponseCommentAttributes> analyzeAndGetAffectedComments() {
+        println("Analyzing comments and their search documents...");
+
+        List<FeedbackResponseCommentAttributes> allComments =
+                FeedbackResponseCommentsLogic.inst().getAllFeedbackResponseComments();
+
+        List<FeedbackResponseCommentAttributes> commentsToFix = new ArrayList<>();
+
+        int numberOfDocuments = 0;
+        int numberOfUnaffectedDocuments = 0;
+        LoopHelper loopHelper = new LoopHelper(batchSize, "comments analyzed.");
+
+        for (FeedbackResponseCommentAttributes comment : allComments) {
+            loopHelper.recordLoop();
+
+            Document document = getDocument(comment);
+            if (document == null) {
+                continue;
+            }
+
+            numberOfDocuments++;
+
+            String sampleDateString = extractSampleDateString(document);
+            if (isInDateFormat(sampleDateString, oldDateFormat)) {
+                commentsToFix.add(comment);
+            } else if (isInDateFormat(sampleDateString, newDateFormat)) {
+                numberOfUnaffectedDocuments++;
+            } else {
+                println("Unrecognised date format (" + sampleDateString + ") for:\n" + comment);
+            }
+        }
+
+        println("\n################# Analysis Results #################");
+        println("Total number of comments: " + loopHelper.getCount());
+        println("Total number of documents: " + numberOfDocuments + "\n");
+
+        println("Number of documents already in new date format: " + numberOfUnaffectedDocuments);
+        println("Number of documents in unrecognizable date format: "
+                + (numberOfDocuments - commentsToFix.size() - numberOfUnaffectedDocuments) + "\n");
+
+        println("Number of documents in old date format: " + commentsToFix.size());
+        println("####################################################\n");
+
+        return commentsToFix;
+    }
+
+    private Document getDocument(FeedbackResponseCommentAttributes comment) {
+        return index.get(comment.getId().toString());
+    }
+
+    private String extractSampleDateString(Document document) {
+        String frcaJson = document.getOnlyField(Const.SearchDocumentField.FEEDBACK_RESPONSE_COMMENT_ATTRIBUTE).getText();
+        return jsonParser.parse(frcaJson).getAsJsonObject().getAsJsonPrimitive("createdAt").getAsString();
+    }
+
+    private synchronized boolean isInDateFormat(String dateString, DateFormat dateFormat) {
```

Comment:
Oh yea this was only needed previously for accessing the `DateFormat`s directly in private helper methods, before the current iteration of `isInDateFormat` was implemented 😅